### ISSUE TITLE
[BUG] HDFS HA name node is not resolved

### ISSUE THREAD

**Issue by jeremyp-bq** [vertica/spark-connector#312]
## Environment
- Vertica Spark Connector version: 3.0.1

---

## Problem Description

When using the Spark Connector against HDFS in HA mode if fails to resolve the name node addresses.

For example, consider the following HDFS config:
```
  <property>
    <name>dfs.namenode.http-address.hanameservice.namenode101</name>
    <value>namenode1.domain:9870</value>
  </property>
  <property>
    <name>dfs.namenode.https-address. hanameservice.namenode101</name>
    <value>namenode1.domain:9871</value>
  </property>
```

Results in the following exception:
```
Issues:
Caused by: com.vertica.spark.util.error.ConnectorException: Could not find name node address in Hadoop configuration. Please set either dfs.namenode.http-address or dfs.namenode.https-address in hdfs-site.xml
```

---

**Comment by jeremyp-bq** [comment]
Refer also to the original ticket for HDFS HA support, #248.

**Comment by jeremyp-bq** [comment]
The error indicates that the connector is trying to find the `dfs.namenode.http-address` or `dfs.namenode.https-address` properties in the `hdfs-site.xml` config file.  So it may be a case of not setting the right properties, or not ensuring those properties are available across Vertica, HDFS, and/or the client.

It is important to note that older versions of Hadoop look for `dfs.http.address` and `dfs.https.address` (See deprecated properties list [here](https://hadoop.apache.org/docs/r2.8.2/hadoop-project-dist/hadoop-common/DeprecatedProperties.html)).  So setting those properties as well guards against using older versions of Hadoop.

By commenting out all of the above properties I was able to cause a Kerberos test to fail, although I did not see the exact same error as indicated in this ticket.

Also note that in the example there is a space in the https entry (`dfs.namenode.https-address. hanameservice.namenode101`).  I;m not sure if this would cause any issues or not.
